examples/tutorials/hybrid_demucs_tutorial.py

```diff
@@ -0,0 +1,352 @@
+            !pip3 install mir_eval
+            """
+        )
+    except ModuleNotFoundError:
+        pass
+    raise
+
+######################################################################
+# 3. Construct the pipeline
+# -------------------------
+# 
+# Pre-trained model weights and related pipeline components are bundled as
+# :py:func:`torchaudio.pipelines.HDEMUCS_HIGH_MUSDB_PLUS`. This is a
+# HDemucs model trained on
+# `MUSDB18-HQ <https://zenodo.org/record/3338373>`__ and additional
+# internal extra training data.
+# This specific model is suited for higher sample rates, around 44.1 kHZ
+# and has a nfft value of 4096 with a depth of 6 in the model implementation.
+
+bundle = HDEMUCS_HIGH_MUSDB_PLUS
+
+model = bundle.get_model()
+
+sample_rate = bundle.sample_rate
+
+print(f"Sample rate: {sample_rate}")
+
+######################################################################
+# 4. Configure the application function
+# -------------------------------------
+# 
+# Because ``HDemucs`` is a large and memory-consuming model it is
+# very difficult to have sufficient memory to apply the model to
+# an entire song at once. To work around this limitation,
+# obtain the separated sources of a full song by
+# chunking the song into smaller segments and run through the
+# model piece by piece, and then rearrange back together.
+# 
+# When doing this, it is important to ensure some
+# overlap between each of the chunks, to accommodate for artifacts at the
+# edges. Due to the nature of the model, sometimes the edges have
+# inaccurate or undesired sounds included.
+# 
+# We provide a sample implementation of chunking and arrangement below. This
+# implementation takes an overlap of 1 second on each side, and then does
+# a linear fade in and fade out on each side. Using the faded overlaps, I
+# add these segments together, to ensure a constant volume throughout.
+# This accommodates for the artifacts by using less of the edges of the
+# model outputs.
+#
+# .. image:: https://download.pytorch.org/torchaudio/tutorial-assets/HDemucs_Drawing.jpg
+
+from torchaudio.transforms import Fade
+
+
+def apply_model(model, mix, segment=10.,
+                overlap=0.1, device=None,
+                ):
+    """
+    Apply model to a given mixture. Use fade, and add segments together in order to add model segment by segment. 
+
+    Args:
+        segment (int): segment length in seconds
+        device (torch.device, str, or None): if provided, device on which to
+            execute the computation, otherwise `mix.device` is assumed.
+            When `device` is different from `mix.device`, only local computations will
+            be on `device`, while the entire tracks will be stored on `mix.device`.
+    """
+    if device is None:
+        device = mix.device
+    else:
+        device = torch.device(device)
+
+    model.to(device)
```

Comment:
This kind of promotion (or operations with "side-effect", generally speaking) should be performed outside of helper function, because it is hard to reason about when viewed from main execution flow. It does not matter much for tutorial, but moving this right after the model creation would make it even easier to follow.